docs/content/posts/2024-10-01-release-2.0-preview.md

```diff
@@ -0,0 +1,106 @@
+
+### Connector Adaption Plan
+
+As SourceFunction, SinkFunction and SinkV1 being removed, existing connectors depending on these APIs will not work on the Flink 2.x series. Here's the plan for adapting the first-party connectors. 
+1. A new version of Kafka connector, adapted to the API changes, will be released right after the release of Flink 2.0 Preview.
+2. JDBC and ElasticSearch connectors will be adapted by the formal release of Flink 2.0.
+3. We plan to gradually migrate the remaining first-party connectors within 3 subsequent minor releases (i.e., by Flink 2.3).
+
+## Configuration
+
+Configuration options meet the following criteria are removed. See also the list of [removed configuration options](#list-of-removed-configuration-options).
+- Annotated as `@Public` and have been deprecated for at least 2 minor releases.
+- Annotated as `@PublicEvolving` and have been deprecated for at least 1 minor releases.
+
+The legacy configuration file `flink-conf.yaml` is no longer supported. Please use `config.yaml` with standard YAML format instead. A migration tool is provided to convert a legacy `flink-conf.yaml` into a new `config.yaml`. See [Migrate from flink-conf.yaml to config.yaml](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/config/#migrate-from-flink-confyaml-to-configyaml) for more details.
+
+Dedicated APIs for configuring specific options are removed from `StreamExecutionEnvironment` and `ExecutionConfig`. All options should now be set via `Configuration` and `ConfigOption`. See also the list of [removed programming APIs](#list-of-removed-programming-apis).
+
+To avoid exposing internal interfaces, User-Defined Functions no longer have full access to `ExecutionConfig`. Instead, necessary functions such as `createSerializer()`, `getGlobalJobParameters()` and `isObjectReuseEnabled()` can now be accessed from `RuntimeContext` directly.
+
+## Misc
+
+**State Compatibility** is not guaranteed between 1.x and 2.x. 
+
+**Java 8** is no longer supported. The minimum Java version supported by Flink now is Java 11.
+
+**Per-Job Deployment Mode** is removed. Please use [Application Mode](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/overview/#application-mode) instead.
+
+**Legacy Mode of Hybrid Shuffle** is removed.
+
+# Highlights of New Features
+
+## Disaggregated State Storage and Management
+
+The past decade has witnessed a dramatic shift in Flink's deployment mode, workload patterns, and hardware improvements. We've moved from the map-reduce era where workers are computation-storage tightly coupled nodes to a cloud-native world where containerized deployments on Kubernetes become standard. To enable Flink's Cloud-Native future, we introduce Disaggregated State Storage and Management that uses Distributed File Systems (DFS) as primary storage in Flink 2.0.
```

Comment:
> Distributed File Systems (DFS)

From my understanding, Object Storage is not a DFS (not a file system). Should we also add "Object Storage" here?

Reply:
Make sense to me. I'll generalize it to "remote storage".